### ISSUE TITLE
Example

### ISSUE THREAD

**Issue by unnamed082** [mlpack/mlpack#2124]
Hi, can you tell me an example of using mlpak for training a neural network (any) with color photos with tags
I don’t understand how to work with color channels, so I want to see an example of working code

**Comment by zoq** [comment]
Hello @unnamed082 it's really similar to: https://github.com/mlpack/models/blob/master/Kaggle/DigitRecognizerCNN/src/DigitRecognizerCNN.cpp, once the data is loaded make sure the channels are vectorized, here is an example:

```cpp
arma::Mat<unsigned char> matrix;
data::ImageInfo info;
data::Load("example.png", matrix, info, false, true);

size_t samples = 1000;
const arma::mat trainX(matrix-n_rows, samples);
trainX.col(0) = arma::vectorise(matrix); // The vectorise isn't necessary since data::load already returns a 1D matrix.
```

afterwards you have to adapt the input channels:

https://github.com/mlpack/models/blob/178ba4aa7c58a7a0bbd54de0118315380212445e/Kaggle/DigitRecognizerCNN/src/DigitRecognizerCNN.cpp#L110

the rest is the same.

**Comment by unnamed082** [comment]
thanks for the quick response, when I run DigitRecognizerCNN.cpp I get an error as I wrote here [but](https://github.com/mlpack/models/issues/35) 
I can attach the whole project

**Comment by unnamed082** [comment]
https://github.com/mlpack/models/issues/35

**Comment by unnamed082** [comment]
@zoq 
please tell me where is the error
I have been trying to deal with this example for a long time :(

**Comment by zoq** [comment]
Ohh, totally missed mlpack/models#35, let me comment on the issue, we probably have to fix that issue first.

**Comment by unnamed082** [comment]
I have to report that today I tried to execute https://github.com/mlpack/models/blob/master/Kaggle/DigitRecognizer/src/DigitRecognizer.cpp
and it worked correctly, it completed
the quality of the solution was 11% because of the line 
https://github.com/mlpack/models/blob/master/Kaggle/DigitRecognizer/src/DigitRecognizer.cpp#L108
but the main thing is that the Ann model worked,
maybe a bug in the network architecture?

**Comment by zoq** [comment]
So, if you remove the dropout layer you get better results? The model in the example is somewhat minimal, you could increase the number of epochs or increase the model size, to get better results.

**Comment by unnamed082** [comment]
yes, when changing the parameters I made the results more than 95%
but the main thing is that this neural network worked, unlike snn

**Comment by unnamed082** [comment]
cnn, sorry
